lib/webpay/WebpayOneClick.php

```diff
@@ -316,8 +319,17 @@ function finishInscription($token) {
      * Permite realizar transacciones de pago. Retorna el resultado de la autorización.
      * Este método que debe ser ejecutado, cada vez que el usuario
      * selecciona pagar con Oneclick
+     *
+     * @throws AmountException si el monto no es válido.
+     * @throws AmountDecimalsException si el monto contiene decimales.
      * */
-    public function authorize($buyOrder, $tbkUser, $username, $amount) {
+    public function authorize($buyOrder, $tbkUser, $username, $amount)
+    {
+        // validaciones $amount
+        if (!is_numeric($amount))
+            throw new AmountException();
+        if (is_float($amount))
```

Comment:
is_float should not work as expected. Sometimes the amout could come from a user input or from a database query on a string variable. 
![image](https://user-images.githubusercontent.com/1103494/78599023-5b0aae00-781e-11ea-8d53-4ce34a5d7f0c.png)

```suggestion
        if (is_float($amount))
```

Reply:
You yousl probably need something like this: 
![image](https://user-images.githubusercontent.com/1103494/78599186-ae7cfc00-781e-11ea-9131-7d4ab6c361d6.png)